core/src/search_stores/search_store.rs

```diff
@@ -78,31 +84,46 @@ const NODES_INDEX_NAME: &str = "core.data_sources_nodes";
 impl SearchStore for ElasticsearchSearchStore {
     async fn index_document(&self, document: &Document) -> Result<()> {
         let node = Node::from(document.clone());
+        self.index_node(node).await
+    }
+
+    async fn index_table(&self, table: &Table) -> Result<()> {
+        let node = Node::from(table.clone());
```

Comment:
We are cloning twice here. Once to pass to `Node::from` and then each field is also cloned in `Node::new`. That's kind of sad. IMO we could do it with 0 clones, but I would be happy with 1 clone :)

Reply:
something like this is IMO better: https://github.com/dust-tt/dust/blob/3a5fcb8a9d72fc7dd440a49d98d10b228dfb593c/core/src/data_sources/node.rs#L70-L79

Reply:
I think index_table should not take ownership of the Table passed, AFAIU -- but i'm only Rust lev 2 so tell me if you know better?

But if you agree, your suggestion would then be to change the  `Node::from` to take references instead of owned values?
Didn't go into that since Table / Document ::from where done & used by others so they might have their reasons, but open to the idea

or were you suggesting something else? glad to have your perspective on this !

Reply:
Something like this instead of from could help https://github.com/dust-tt/dust/blob/3a5fcb8a9d72fc7dd440a49d98d10b228dfb593c/core/src/data_sources/node.rs#L70-L79

Reply:
I also think it's fine if `index_table` takes ownership. If you want to keep your table, you can clone it before you pass it to `index_table`. And if you don't need it you can just move it. 
At least you have the choice. IMHO, taking a ref just to clone immediately is never good, because you remove the optionality from the caller (you are forcing a clone even in situations where a clone might not be needed)

Reply:
All right; good point for `index_table` taking ownership, will change
For into_folder, i'd rather keep the trait with From at the time since this trait seems standard
but we'll be down to 1 clone :+1: